src/credentials/activation.ts

```diff
@@ -77,32 +83,44 @@ export async function loginWithMostRecentCredentials(
         if (await tryConnect(loginCredentialsId, false)) {
             return
         }
-        getLogger().warn('autoconnect: failed to login "%s"', previousCredentialsId)
+        getLogger().warn('autoconnect: login failed: "%s"', previousCredentialsId)
     }
 
     const providerMap = await manager.getCredentialProviderNames()
     const profileNames = Object.keys(providerMap)
     // Look for "default" profile or exactly one (any name).
-    const defaultProfile = profileNames.includes('profile:default')
-        ? 'profile:default'
+    const defaultProfile = profileNames.includes(defaultName)
+        ? defaultName
         : profileNames.length === 1
         ? profileNames[0]
         : undefined
 
-    if (!previousCredentialsId && !defaultProfile) {
+    if (!previousCredentialsId && profileNames.length === 0) {
         await loginManager.logout(true)
         getLogger().info('autoconnect: skipped (profileNames=%d)', profileNames.length)
         return
     }
 
-    // Auto-connect if there is a default profile.
+    // Try to auto-connect the default profile.
     if (defaultProfile) {
-        getLogger().debug('autoconnect: trying "%s"', defaultProfile)
+        getLogger().info('autoconnect: trying "%s"', defaultProfile)
         if (await tryConnect(providerMap[defaultProfile], !isCloud9())) {
             return
         }
     }
 
+    // Try to auto-connect up to 3 other profiles (useful for Cloud9, ECS, …).
+    for (let i = 0; i < 4 && i < profileNames.length; i++) {
```

Comment:
TODO:
Seems like we could just sort the profiles recent -> default -> rest (and change the upper bound to 6) then remove the above logic right?

Unless we want to handle the first two cases differently, though right now it looks the same. We also already store the 5 most recently used profiles, so we could get a bit more value out of `CredentialsProfileMru`. Makes sense to prioritize those.